Add a support for PHAR building

lyrixx opened the pull request:
fix #54
fix #63
fix #62

welcoMattic commented on README.md:
```diff
@@ -55,6 +55,13 @@ $fixer = new Fixer(array("Trademark", "SmartQuotes"));
+
+You can run a standalone version of JoliTypo by downloading [the PHAR version](https://github.com/jolicode/JoliTypo/releases/latest)
+
+Run `jolitypo --help` to know how to configure the Fixer.
```
```suggestion
Run `jolitypo.phar --help` to know how to configure the Fixer.
```

lyrixx replied:
Nope. the suffix is useless to me 
(Like composer)

welcoMattic commented on tools/phar/README.md:
````diff
@@ -0,0 +1,8 @@
+
+```bash
+composer install
+vendor/bin/box  compile
````
```suggestion
vendor/bin/box compile
```

welcoMattic commented on tools/phar/composer.json:
```diff
@@ -0,0 +1,30 @@
+    },
+    "autoload": {
+        "psr-4": {
+            "Jolicode\\JolitypoPhar\\": "src/"
```
```suggestion
            "JoliCode\\JolitypoPhar\\": "src/"
```

welcoMattic commented:
https://github.com/jolicode/JoliTypo/blob/039246f9e2b16c530bfc99056d8688561b2592f7/tools/phar/a.html should be removed, no?

welcoMattic commented on tools/phar/composer.json:
```diff
@@ -0,0 +1,30 @@
+        {
+            "name": "Grégoire Pineau",
+            "email": "lyrixx@lyrixx.info"
+        }
```
```suggestion
        },
        {
            "name": "Mathieu Santostefano",
            "email": "msantostefano@pm.me"
        }
```

lyrixx commented:
@welcoMattic I addressed your comment 👍🏼

damienalexandre reviewed (approved):
A note in the Changelog could be added too. 

I'm not a big fan of the hardcoded `$recommendedRulesByLocale` also. That should either be in the main library, or nowhere IMO.

lyrixx commented:
> I'm not a big fan of the hardcoded `$recommendedRulesByLocale` also. That should either be in the main library, or nowhere IMO.

I moved it to the main lib, and I also added the phar compilation in the CI
